src/main/java/uk/gov/hmcts/ccd/domain/service/casedataaccesscontrol/RoleAssignmentService.java

```diff
@@ -34,6 +46,46 @@ public RoleAssignmentService(@Qualifier(CachedRoleAssignmentRepository.QUALIFIER
         this.roleAssignmentsFilteringService = roleAssignmentsFilteringService;
     }
 
+    public RoleAssignments createCaseRoleAssignments(final CaseDetails caseDetails,
+                                                     final String userId,
+                                                     final Set<String> roles,
+                                                     final boolean replaceExisting) {
+
+        RoleRequestResource roleRequest = RoleRequestResource.builder()
+            .assignerId(userId)
+            .process(RoleAssignmentRepository.DEFAULT_PROCESS)
+            .reference(createRoleRequestReference(caseDetails, userId))
+            .replaceExisting(replaceExisting)
+            .build();
+
+        List<RoleAssignmentResource> requestedRoles = roles.stream()
+            .map(roleName -> RoleAssignmentResource.builder()
+                .actorIdType(ActorIdType.IDAM.name())
+                .actorId(userId)
+                .roleType(RoleType.CASE.name())
+                .roleName(roleName)
+                .classification(Classification.RESTRICTED.name())
+                .grantType(GrantType.SPECIFIC.name())
+                .roleCategory(RoleCategory.PROFESSIONAL.name())
```

Comment:
Note: https://github.com/hmcts/ccd-data-store-api/pull/1552 has been merged into the base branch so should be available now for use 👍

Reply:
Resolved in commits: 
* f27310cf47502f7539c4967a5f691f5ba2bd15ef merge
* 9edd20c093d4f743ed4b1b47be6fd17e068c6d32 refactor
* 4855ab8f52a07dfd69ab25d0ae8dc91de96de7ee integrate with RDM-12424 from #1552